app/assets/javascripts/discourse/app/components/emoji-picker.js

```diff
@@ -27,40 +26,39 @@ function customEmojis() {
   return groups;
 }
 
-export default Component.extend({
-  emojiStore: service("emoji-store"),
-  tagName: "",
-  customEmojis: null,
-  recentEmojis: null,
-  hoveredEmoji: null,
-  isActive: false,
-  usePopper: true,
-  placement: "auto", // one of popper.js' placements, see https://popper.js.org/docs/v2/constructors/#options
-  initialFilter: "",
-  elements: {
+@tagName("")
+export default class EmojiPicker extends Component {
+  @service("emoji-store") emojiStore;
+
+  customEmojis = customEmojis();
+  recentEmojis = null;
+  hoveredEmoji = null;
+  isActive = false;
+  usePopper = true;
+  placement = "auto"; // one of popper.js' placements, see https://popper.js.org/docs/v2/constructors/#options
+  initialFilter = "";
+
+  elements = {
     searchInput: ".emoji-picker-search-container input",
     picker: ".emoji-picker-emoji-area",
-  },
+  };
 
   init() {
-    this._super(...arguments);
-    this.set("customEmojis", customEmojis());
-    if ("IntersectionObserver" in window) {
-      this._sectionObserver = this._setupSectionObserver();
-    }
-  },
+    super.init(...arguments);
+    this._sectionObserver = this._setupSectionObserver();
```

Comment:
can we move this out of the `init`?

Reply:
Feels weird to put this kind of side-effect-y thing in the field initializer... so I moved it to didInsertElement instead. Makes sense since it's DOM-related 👌 

(also BTW, this whole file is removed in https://github.com/discourse/discourse/pull/28277, so this is just a temporary thing to remove the last classic-class-syntax component from core)